### ISSUE TITLE
Unable to install 5.2.0 on Windows 10 professional

### ISSUE THREAD

**Issue by donzelch** [eXist-db/exist#3276]
Launcher fails launching because a path or the command itself is too long.

Tried  alternative installations : unable anyway to initialize any database.

**Comment by adamretter** [comment]
Can you give us some more detail? e.g. the paths involved and any relevant error messages/logs?

**Comment by donzelch** [comment]
The following line, on Windows 10 doesn't work if the BASEDIR path is longer than 15 characters.

%JAVACMD% %JAVA_OPTS% -Xms128m -Dfile.encoding=UTF-8 -Dlog4j.configurationFile="%BASEDIR%"/etc/log4j2.xml -Dexist.home="%BASEDIR%" -Dexist.configurationFile="%BASEDIR%"/etc/conf.xml -Djetty.home="%BASEDIR%" -Dexist.jetty.config="%BASEDIR%"/etc/jetty/standard.enabled-jetty-configs -classpath %CLASSPATH% -Dapp.name="launcher" -Dapp.repo="%REPO%" -Dapp.home="%BASEDIR%" -Dbasedir="%BASEDIR%" org.codehaus.mojo.appassembler.booter.AppassemblerBooter %CMD_LINE_ARGS%

This is probably due to the length of the path in some jar file. I solved it by copying the installation folder into the root folder.

**Comment by adamretter** [comment]
@donzelch Did your BASEDIR path previously have spaces in it?

**Comment by donzelch** [comment]
No. It was G:\Extrasoft\existdb_5.2.0. After initializing the platform, I moved it to the original place and everything is working fine. I had previously activated the "long file names" option of Windows 10 because I know that long path into jar files often create problems when copying Eclipse based RCP applications. On Windows 7, using 7zip for moving a RCP application from one disk to another was quite mandatory.

So may be, the problem is just with the length of the command itself. I haven't tried, for example, to rename the "launcher.bat" into "cmd.bat". Sometime Microsoft is keeping old interfaces for compatibility reasons.

**Comment by adamretter** [comment]
@donzelch not quite sure what we can do here. Windows does have a limitation on the command line length when executing from `cmd.exe`. For that exact reason we previously had to adopt `AppassemblerBooter`, so that it would load the classpath dynamically for us rather than it becoming very long.

**Comment by adamretter** [comment]
@donzelch Do you think that providing PoweShell alternatives to the .bat files would solve this problem?

**Comment by adamretter** [comment]
See https://github.com/mojohaus/appassembler/pull/70

**Comment by peta** [comment]
Negative, I wrote a PowerShell script using `Start-Process` to invoke the javaw command line (quite long due to class paths), but did not work either. Then just moved the existdb folder up to `C:\existdb`, no elegant solution but works.
